common/web/types/src/ldml-keyboard/ldml-keyboard-xml-reader.ts

```diff
@@ -21,6 +21,17 @@ export default class LDMLKeyboardXMLSourceFileReader {
     boxXmlArray(source?.keyboard?.displays, 'display');
     boxXmlArray(source?.keyboard?.names, 'name');
     boxXmlArray(source?.keyboard?.vkeys, 'vkey');
+    // for implied we need keyboard.keys.import to exist.
+    if (!source?.keyboard?.keys) {
+      source.keyboard.keys = {
+        import:[],
+        key:[],
+        flicks:[],
+      };
+    }
+    if (!source?.keyboard?.keys?.import) {
+      source.keyboard.keys.import = [];
```

Comment:
See above -- we need to construct the whole object tree if we want to do this.

Is this the best place for it? :grin: